projects/core/components/textfield/textfield.directive.ts

```diff
@@ -67,8 +67,13 @@ export class TuiTextfieldBase<T> implements OnChanges {
     }
 
     public setValue(value: T | null): void {
-        this.el.value = value == null ? '' : this.textfield.stringify(value);
-        this.el.dispatchEvent(new Event('input', {bubbles: true}));
+        this.el.focus();
+        this.el.select();
+        this.el.ownerDocument.execCommand(
```

Comment:
what about `execCommand is deprecated`?
<img width="826" alt="image" src="https://github.com/user-attachments/assets/d3126d42-d323-463e-b48a-71e459d92bed">

Reply:
It's not going away anytime soon, no other way to kick Safari to work properly